yaml-tests/src/main/java/com/apple/foundationdb/relational/yamltests/command/QueryExecutor.java

```diff
@@ -170,6 +190,62 @@ private Continuation executeContinuation(@Nonnull RelationalConnection connectio
         return continuationAfter;
     }
 
+    private RelationalPreparedStatement prepareContinuationStatement(@Nonnull RelationalConnection connection,
+                                                                     @Nonnull Continuation continuation,
+                                                                     @Nullable Integer maxRows) throws SQLException {
+        var s = connection.prepareStatement("EXECUTE CONTINUATION ?;");
+        s.setMaxRows((maxRows != null) ? maxRows : DEFAULT_MAX_ROWS);
+        if (parameters != null) {
+            setParametersInPreparedStatement(s, connection);
+        }
+        // set continuation
+        s.setBytes(1, continuation.serialize());
+        return s;
+    }
+
+    private Object executeStatementAndForceContinuations(@Nonnull Statement s, @Nullable String queryString,
+                                                         final RelationalConnection connection, @Nullable Integer maxRows) throws SQLException {
+        // Check if we need to force continuations
+        if ((maxRows == null) && forceContinuations) {
+            s.setMaxRows(FORCED_MAX_ROWS);
+            Object result = executeStatement(s, queryString);
+            if (result instanceof RelationalResultSet) {
+                List<RelationalResultSet> results = new ArrayList<>();
+                @SuppressWarnings("PMD.CloseResource")
+                RelationalResultSet resultSet = (RelationalResultSet)result;
+                RelationalResultSetMetaData metadata = resultSet.getMetaData(); // The first metadata will be used for all
+                boolean hasResult = resultSet.next(); // Initialize result set value retrieval. Has only one row.
+                // Edge case: when there are no results at all, we don't want to store the result set at all,
+                // to create an empty aggregated result set
+                if (hasResult) {
+                    results.add(resultSet);
+                }
+                // Have continuations - keep running the query
+                Continuation continuation = resultSet.getContinuation();
+                while (!continuation.atEnd()) {
```

Comment:
It seems like something here should assert that the number returned by each individual result is `<= FORCED_MAX_ROWS`

Reply:
That is a good addition. Unfortunately, we can't do It at the result level (here) because of the statefulness of the result set interface. I tried to do that within the aggregate result set but since we can't control who calls `next()` on the individual rows, we can't assert on their status. Ideas welcome.